craft_parts/features.py

```diff
@@ -0,0 +1,39 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2022 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Global features to be configured by the application."""
+
+import contextlib
+import dataclasses
+import logging
+
+from craft_parts.utils import Singleton
+
+logger = logging.getLogger()
+
+
+@dataclasses.dataclass(frozen=True)
+class Features(metaclass=Singleton):
+    """Configurable craft-parts features."""
+
+    enable_overlay: bool = True
```

Comment:
so do we have an opinion on `overlays` specifically being default-enabled vs default-disabled? I ask because of the applications we have, only Rockcraft uses it so everyone *else* has to disable it.

Reply:
Disabled by default seems like it would be the safer option since overlays require the application to do additional work outside of what `craft-parts` provides.

Reply:
We can change that to defaut-disabled, but rockcraft will need to enable it explicitly.